### PR TITLE
User Resource Ubuntu update 

### PR DESCRIPTION
**Pull request by wojciechcloudkubed**
Addresses [this issue ](https://github.com/microsoft/AzureTRE/issues/3523)

- [x] Nexus
- [x] Resource Processor
- [x] Guacamole VMs
- [x] Readme and docs
- [ ] Test

### CODE REVIEW

**Comment by wojciechcloudkubed** [comment]
@microsoft-github-policy-service agree company="CloudKubed"

File: templates/workspace_services/guacamole/user_resources/README.md [hunk 1]
```diff
@@ -29,21 +28,29 @@ custom:
     "8 CPU | 32GB RAM": Standard_D8s_v5
     "16 CPU | 64GB RAM": Standard_D16s_v5
   image_options:
-    "Ubuntu 18.04":
+    "Ubuntu 22.04 LTS":
```
Comment (SvenAelterman) [inline templates/workspace_services/guacamole/user_resources/README.md:31]: I think we should include the new `apt-sku` property here as well.

File: templates/shared_services/sonatype-nexus-vm/terraform/vm.tf [hunk 2]
```diff
@@ -112,7 +113,7 @@ resource "azurerm_linux_virtual_machine" "nexus" {
   source_image_reference {
     publisher = "Canonical"
     offer     = "0001-com-ubuntu-server-jammy"
-    sku       = "22_04-lts-gen2"
+    sku       = "22_04-lts"
```
Comment (SvenAelterman) [inline templates/shared_services/sonatype-nexus-vm/terraform/vm.tf:116]: What's the reason for not using `gen2`?

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/porter.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 schemaVersion: 1.0.0
 name: tre-service-guacamole-linuxvm
-version: 0.6.9
+version: 0.6.17
```
Comment (SvenAelterman) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/porter.yaml:4]: I am not sure if semantic versioning is used for these bundles, but it feels like this should be at least a minor version increase, if not a major.
(E.g., how will the Update command work?)

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/linuxvm.tf [hunk 4]
```diff
@@ -131,6 +131,7 @@ data "template_file" "apt_sources_config" {
   template = file("${path.module}/apt_sources_config.yml")
   vars = {
     nexus_proxy_url = local.nexus_proxy_url
+    sku             = local.apt_sku
```
Comment (SvenAelterman) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/linuxvm.tf:134]: I am concerned that the inconsistency between using `apt`, `apt-sku` and `apt_sku` is going to make maintenance harder.

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/porter.yaml (hunk 3 above)
Comment (tim-allen-ck) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/porter.yaml]: agreed will change then to 1.0.0 when I've finished testing

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/linuxvm.tf (hunk 4 above)
Comment (tim-allen-ck) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/linuxvm.tf:134]: agreed, I'll change it to something more consistent

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh
[the remark this answers is not in the document]
Comment (Danny-Cooke-CK) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh:1]: interesting. so i wonder if running something like this from within the bootstrap will work 

su - username -c "command;command;command"

[the remark this answers is not in the document]
Comment (SvenAelterman) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh:1]: Indeed, it does. This is what my line of code looks like (I am not a Linux expert, took my several tries to get it right. There might be a better way):
`sudo -u ${USERNAME} -i bash -c 'echo xfce4-session > ~/.xsession'`

[the remark this answers is not in the document]
Comment (SvenAelterman) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh:1]: Also, noting that I had to do this even for the DSVM marketplace image where the UI is already installed.

[the remark this answers is not in the document]
Comment (Danny-Cooke-CK) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh:1]: @SvenAelterman could you possibly share your bootstrap script please?

[the remark this answers is not in the document]
Comment (SvenAelterman) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh:1]: Yes, but be aware I modified it only for the DSVM Ubuntu 20 image. I did not get around to updating for plain Ubuntu 20/22.
Also, some of the commented out lines at the bottom might be necessary but I was just trying to get the basics to work.
https://gist.github.com/SvenAelterman/4771b6accf1fe82ac782e5c0e00efa0b

**Pull request closed without merging**